src/test/py/bazel/bzlmod/bazel_lockfile_test.py

```diff
@@ -964,6 +964,46 @@ def testExtensionEvaluationDoesNotRerunOnChangedImports(self):
         stderr,
     )
 
+  def testLockfileRecreatedAfterDeletion(self):
+    self.ScratchFile(
+      'MODULE.bazel',
+      [
+        'lockfile_ext = use_extension("extension.bzl", "lockfile_ext")',
+        'use_repo(lockfile_ext, "hello")',
+      ],
+    )
+    self.ScratchFile('BUILD.bazel')
+    self.ScratchFile(
+      'extension.bzl',
+      [
+        'def _repo_rule_impl(ctx):',
+        '    ctx.file("WORKSPACE")',
+        '    ctx.file("BUILD", "filegroup(name=\'lala\')")',
+        '',
+        'repo_rule = repository_rule(implementation=_repo_rule_impl)',
+        '',
+        'def _module_ext_impl(ctx):',
+        '    repo_rule(name="hello")',
+        '',
+        'lockfile_ext = module_extension(',
+        '    implementation=_module_ext_impl,',
+        ')',
+      ],
+    )
+
+    _, _, _ = self.RunBazel(['build', '@hello//:all'])
```

Comment:
No need for this.

Reply:
Wasn't sure whether this would trigger a linter. Feel free to remove on import (or let me know that I should drop it here).